Formula/linode-cli.rb

```diff
@@ -0,0 +1,88 @@
+class LinodeCli < Formula
+  include Language::Python::Virtualenv
+
+  desc "CLI for the Linode API"
+  homepage "https://www.linode.com/products/cli/"
+  url "https://github.com/linode/linode-cli/archive/refs/tags/5.14.0.tar.gz"
+  sha256 "64ec447910eda75759e88e814846432b75ff08211ff1d1a86b223d5db5cd9902"
+  license "BSD-3-Clause"
+  head "https://github.com/linode/linode-cli.git", branch: "master"
+
+  depends_on "poetry" => :build # for terminaltables
+  depends_on "openssl@1.1"
+  depends_on "python@3.10"
+
+  resource "certifi" do
+    url "https://files.pythonhosted.org/packages/6c/ae/d26450834f0acc9e3d1f74508da6df1551ceab6c2ce0766a593362d6d57f/certifi-2021.10.8.tar.gz"
+    sha256 "78884e7c1d4b00ce3cea67b44566851c4343c120abd683433ce934a68ea58872"
+  end
+
+  resource "charset-normalizer" do
+    url "https://files.pythonhosted.org/packages/48/44/76b179e0d1afe6e6a91fd5661c284f60238987f3b42b676d141d01cd5b97/charset-normalizer-2.0.10.tar.gz"
+    sha256 "876d180e9d7432c5d1dfd4c5d26b72f099d503e8fcc0feb7532c9289be60fcbd"
+  end
+
+  resource "idna" do
+    url "https://files.pythonhosted.org/packages/62/08/e3fc7c8161090f742f504f40b1bccbfc544d4a4e09eb774bf40aafce5436/idna-3.3.tar.gz"
+    sha256 "9d643ff0a55b762d5cdb124b8eaa99c66322e2157b69160bc32796e824360e6d"
+  end
+
+  resource "PyYAML" do
+    url "https://files.pythonhosted.org/packages/36/2b/61d51a2c4f25ef062ae3f74576b01638bebad5e045f747ff12643df63844/PyYAML-6.0.tar.gz"
+    sha256 "68fb519c14306fec9720a2a5b45bc9f0c8d1b9c72adf45c37baedfcd949c35a2"
+  end
+
+  resource "requests" do
+    url "https://files.pythonhosted.org/packages/60/f3/26ff3767f099b73e0efa138a9998da67890793bfa475d8278f84a30fec77/requests-2.27.1.tar.gz"
+    sha256 "68d7c56fd5a8999887728ef304a6d12edc7be74f1cfa47714fc8b414525c9a61"
+  end
+
+  resource "terminaltables" do
+    url "https://files.pythonhosted.org/packages/f5/fc/0b73d782f5ab7feba8d007573a3773c58255f223c5940a7b7085f02153c3/terminaltables-3.1.10.tar.gz"
+    sha256 "ba6eca5cb5ba02bba4c9f4f985af80c54ec3dccf94cfcd190154386255e47543"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/b0/b1/7bbf5181f8e3258efae31702f5eab87d8a74a72a0aa78bc8c08c1466e243/urllib3-1.26.8.tar.gz"
+    sha256 "0e7c33d9a63e7ddfcb86780aac87befc2fbddf46c58dbb487e0855f7ceec283c"
+  end
+
+  def install
+    venv = virtualenv_create(libexec, "python3", system_site_packages: false)
+    resource("terminaltables").stage do
+      system Formula["poetry"].opt_bin/"poetry", "build", "--format", "wheel", "--verbose", "--no-interaction"
+      venv.pip_install Dir["dist/terminaltables-*.whl"].first
+    end
+    venv.pip_install resources.reject { |r| r.name == "terminaltables" }
+
+    # Manually download the spec using curl, since the bake command may fail if it enounters an HTTP redirect
+    system "curl", "-L", "-o", "openapi.yaml", "https://www.linode.com/docs/api/openapi.yaml"
```

Comment:
Usually we don't want to download an unversioned/modified-in-place file.

May want to consider a `resource` using a versioned URL, like the raw file using a tag from GitHub: https://github.com/linode/linode-api-docs/tags.

Not sure if specific version is important. Either:
- Version of API listed on GitHub release, e.g. https://github.com/linode/linode-cli/releases/tag/5.14.0 - **API spec version 4.112.0**
- Latest version of API available at time, e.g. **4.113.0**

---

As reference from other package managers:
- NixOS is downloading **4.112.0** API with **5.14.0** release https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/virtualization/linode-cli/default.nix#L14-L21
- Fedora Rawhide keeps a local copy and has used **4.111.0** API with **5.13.2** release.

Reply:
The linode-cli project uses the latest version of the spec at the time of each release. Ideally, we should be doing the same.

If we were to use a specific version of the spec in the formula, how would we ensure that the spec get updated whenever a new one is available? Or at least ensure that it gets updated every once in a while.

Is there a way, for example, to livecheck the [linode-api-docs release page](https://github.com/linode/linode-api-docs/releases) and automatically update the spec version number in the formula if there's an update?

Reply:
If it's a resource people will be reminded to check for updates to the resource when they make a pull request.

Reply:
Here's an example of one of those messages:

https://github.com/Homebrew/homebrew-core/pull/93828#issue-1114891345

Users will also see a message like that after running the `brew bump-formula-pr` command in the terminal.

Reply:
Ok, that's good to hear. I just pushed the new changes.